halibot/message.py

```diff
@@ -26,12 +26,26 @@ def makeOrigin(self):
 
 class Message():
 
+	class Author(str):
+		def __init__(self, a, id=""):
+			self.id = id
+			self.name = str(a) if a else ""
+		def __str__(self):
+			return self.name
+		def __repr__(self):
+			return self.name
+		def __setattr__(self, key, value):
+			if key in ("id", "name"):
+				object.__setattr__(self, key, value)
```

Comment:
Custom data in the `author` object is similar to that of what `misc` would have been then, it would be agent-specific information that modules would need to actively look for (which are not defined in any specification anywhere). I'm fine with removing the checks, but I don't like the idea of encouraging extra metadata hiding in random places.

`__getattr__` is needed, otherwise tests fail.